ecommerce/views/v0/__init__.py

```diff
@@ -199,6 +207,36 @@ def get_queryset(self):
         return BasketDiscount.objects.filter(redeemed_basket__user=self.request.user)
 
 
+class DiscountViewSet(ModelViewSet):
+    """API view set for Discounts"""
+
+    serializer_class = DiscountSerializer
+    queryset = Discount.objects.all()
+    authentication_classes = (SessionAuthentication, TokenAuthentication)
+    permission_classes = (IsAuthenticated, IsAdminUser)
+
+    @action(
+        detail=True,
+        methods=["get"],
+        name="List Discount Redemptions",
+        url_name="redemptions",
+    )
+    def redemptions(self, request, pk):
```

Comment:
Could you use `rest_framework_extensions.routers.NestedRouterMixin` with a dedicated readonly view instead of a single detail view so that both list and detail for the redemptions?